OutOfSchool/OutOfSchool.WebApi/Controllers/V2/WorkshopController.cs

```diff
@@ -146,18 +161,46 @@ public async Task<IActionResult> GetByFilter([FromQuery] WorkshopFilter filter,
     [Consumes("multipart/form-data")]
     public async Task<IActionResult> Create([FromForm] WorkshopV2Dto dto)
     {
-        var userHasRights = await IsUserProvidersOwner(dto.ProviderId).ConfigureAwait(false);
+        if (dto == null)
+        {
+            return BadRequest("Workshop is null.");
+        }
+
+        if (await IsProviderBlocked(dto.ProviderId).ConfigureAwait(false))
+        {
+            return StatusCode(403, "Forbidden to create workshops at blocked providers");
+        }
+
+        if (await IsCurrentUserBlocked())
+        {
+            return StatusCode(403, "Forbidden to create the workshop by the blocked provider.");
+        }
+
+        if (!ModelState.IsValid)
```

Comment:
should not this check be performed before checks for blocked users ?

Reply:
I assumed that check - 'is user blocked?' would take less time than validation of Model. That's why I chose this order. Is this wrong?